pkg/containerwatcher/v1/container_watcher.go

```diff
@@ -321,8 +326,11 @@ func CreateIGContainerWatcher(cfg config.Config, applicationProfileManager appli
 		if event.K8s.ContainerName == "" {
 			return
 		}
+		k8sContainerID := utils.CreateK8sContainerID(event.K8s.Namespace, event.K8s.PodName, event.K8s.ContainerName)
```

Comment:
we usually compute k8sContainerID inside the callbacks (see the managers) to avoid computing it if we disable the feature

Reply:
I see that we do it in `container_watcher`  for almost every tracer.
It will require another interface to do it otherwise and I did not want to do it